### PR TITLE
feat: Console mark medication taken

### LINKED ISSUE #19: Mark Taken Status for Daily Medication
## User Story
As a user, I would like to indicate that I have taken a medication so that:
- I know when to take my next dose.
- I can keep track of how well I am following my doctors orders.
- I can show the doctor that I am taking my medications at the appropriate times of the day.

## Tasks
- [x] Low level code on implementing the data structures and methods to interact with (#20)
- [x] Command parsing to obtain user input (#40)
- [x] Connect user input with low level code (#54)

### PR DESCRIPTION
**Pull request by wenenhoe**
# Checklist for feature implementation and modification
Closes #19 
- [x] Added relevant milestone (under milestone tab on the right)
- Implementation
  - [ ] Sources (if any) credited under both relevant section(s) and Developer Guide (Acknowledgement)
  - [x] JUnit Test(s)
- Documentation
  - [x] JavaDoc for classes and methods
  - Developer Guide
    - [ ] Design (Can add as a todo statement for later on)
    - [ ] Implementation (Can add as a todo statement for later on)
    - [ ] Relevant User Story present
    - [ ] Non-functional requirement(s) that the feature addresses (if any)
    - [ ] Glossary of the (technical) term(s) used (if any)
    - [ ] Specialised instructions for manual testing (if any)
  - User Guide
    - [ ] Description and usage of feature (if any)
  - Contributions (under `/docs/team/*.md`)
    - [ ] Code/Feature/Enhancement contribution
    - [ ] Developer Guide contribution
    - [ ] User Guide contribution
    - [ ] Misc and team-based admin tasks (if any)

# Checklist for non-implementation related PR
Closes #<Issue_Number>
- [ ] Added relevant milestone (under milestone tab on the right)
- Documentation (Contributions)
  - [ ] Developer Guide contributions (if any)
  - [ ] User Guide contributions (if any)
  - [ ] Review contributions (if any)
  - [ ] Miscellaneous and team-based admin tasks (if any)

# Additional Information

Nil

### CODE REVIEW

File: src/main/java/meditracker/command/TakeCommand.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import meditracker.argument.ArgumentList;
+import meditracker.argument.ArgumentName;
+import meditracker.argument.ListIndexArgument;
+import meditracker.exception.ArgumentNotFoundException;
+import meditracker.medication.MedicationManager;
+import meditracker.ui.Ui;
+
+import java.util.Map;
+
+public class TakeCommand extends Command {
```
Comment (T0nyLin) [inline src/main/java/meditracker/command/TakeCommand.java:13]: Rmb to add JavaDoc for the class

File: src/main/java/meditracker/command/UntakeCommand.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import meditracker.argument.ArgumentList;
+import meditracker.argument.ArgumentName;
+import meditracker.argument.ListIndexArgument;
+import meditracker.exception.ArgumentNotFoundException;
+import meditracker.medication.MedicationManager;
+import meditracker.ui.Ui;
+
+import java.util.Map;
+
+public class UntakeCommand extends Command {
```
Comment (T0nyLin) [inline src/main/java/meditracker/command/UntakeCommand.java:13]: Rmb to add JavaDoc for the class

**Review by T0nyLin — COMMENTED** [review]
LGTM, left to include JavaDoc for mark and unmark command classes

File: src/main/java/meditracker/command/TakeCommand.java (hunk 1 above)
Comment (wenenhoe) [reply src/main/java/meditracker/command/TakeCommand.java:17]: Resolved in 1c1c392

File: src/main/java/meditracker/command/UntakeCommand.java (hunk 2 above)
Comment (wenenhoe) [reply src/main/java/meditracker/command/UntakeCommand.java:17]: Resolved in 1c1c392

**Pull request merged**